<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Random Walk with Restart on Multiplex and Heterogeneous Network â€¢ RandomWalkRestartMH</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Random Walk with Restart on Multiplex and Heterogeneous Network">
<meta property="og:description" content="RandomWalkRestartMH">
<meta property="og:image" content="/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">RandomWalkRestartMH</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/RandomWalkRestartMH.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/alberto-valdeolivas/RandomWalkRestartMH/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://twitter.com/alvaldeolivas">
    <span class="fab fa fab fa-twitter fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="https://www.marseille-medical-genetics.org/a-baudot/">
    <span class="fas fa fas fa-university"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="RandomWalkRestartMH_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Random Walk with Restart on Multiplex and Heterogeneous Network</h1>
                        <h4 class="author">Alberto Valdeolivas</h4>
            <address class="author_afil">
      MMG, Marseille Medical Genetics U 1251, Faculte de Medecine, France<br><a class="author_email" href="mailto:#"></a><a href="mailto:alvaldeolivas@gmail.com" class="email">alvaldeolivas@gmail.com</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/alberto-valdeolivas/RandomWalkRestartMH/blob/master/vignettes/RandomWalkRestartMH.Rmd"><code>vignettes/RandomWalkRestartMH.Rmd</code></a></small>
      <div class="hidden name"><code>RandomWalkRestartMH.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>This vignette describes how to use the <em><a href="https://bioconductor.org/packages/3.11/RandomWalkRestartMH">RandomWalkRestartMH</a></em> package to run Random Walk with Restart algorithms on monoplex, multiplex, heterogeneous, multiplex-heterogeneous networks and full multiplex-heterogeneous networks It is based on the work we presented on the following article:</p>
      <p><a href="https://academic.oup.com/bioinformatics/article/35/3/497/5055408" class="uri">https://academic.oup.com/bioinformatics/article/35/3/497/5055408</a></p>
      <p>Although, we have recently extended the method to take into account weigthed networksand full multiplex-heterogeneous networks (both networks connected by bipartite interactions are multiplex.)</p>
    </div>
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p><em><a href="https://bioconductor.org/packages/3.11/RandomWalkRestartMH">RandomWalkRestartMH</a></em> (Random Walk with Restart on Multiplex and Heterogeneous Networks) is an <em>R</em> package built to provide an easy interface to perform Random Walk with Restart in different types of complex networks:</p>
<ol style="list-style-type: decimal">
<li>Monoplex networks (Single networks).</li>
<li>Multiplex networks.</li>
<li>Heterogeneous networks.</li>
<li>Multiplex-Heterogeneous networks.</li>
</ol>
<p>It is based on the work we presented in the article:</p>
<p><a href="https://academic.oup.com/bioinformatics/article/35/3/497/5055408" class="uri">https://academic.oup.com/bioinformatics/article/35/3/497/5055408</a></p>
<p>We have recently extended the method in order to take into account weighted networks. In addition, the package is now able to perform Random Walk with Restart on:</p>
<ol start="5" style="list-style-type: decimal">
<li>Full multiplex-heterogeneous networks.</li>
</ol>
<p>RWR simulates an imaginary particle that starts on a seed(s) node(s) and follows randomly the edges of a network. At each step, there is a restart probability, <code>r</code>, meaning that the particle can come back to the seed(s) <span class="citation">(Pan et al. 2004)</span>. This imaginary particle can explore the following types of networks:</p>
<ul>
<li><p>A <strong>monoplex or single network</strong>, which contains solely nodes of the same nature. In addition, all the edges belong to the same category.</p></li>
<li><p>A <strong>multiplex network</strong>, defined as a collection of monoplex networks considered as layers of the multiplex network. In a multiplex network, the different layers share the same set of nodes, but the edges represent relationships of different nature <span class="citation">(Battiston, Nicosia, and Latora 2014)</span>. In this case, the RWR particle can jump from one node to its counterparts on different layers.</p></li>
<li><p>A <strong>heterogeneous network</strong>, which is composed of two monoplex networks containing nodes of different nature. These different kind of nodes can be connected thanks to bipartite edges, allowing the RWR particle to jump between the two networks.</p></li>
<li><p>A <strong>multiplex and heterogeneous network</strong>, which is built by linking the nodes in every layer of a multiplex network to nodes of different nature thanks to bipartite edges.</p></li>
<li><p>A <strong>full multiplex and heterogeneous network</strong>, in which the two networks connected by bipartite interactions are of multiplex nature. The RWR particle can now explore the full multiplex-heterogeneous network.</p></li>
</ul>
<p>The user can integrate single networks (monoplex networks) to create a multiplex network. The multiplex network can also be integrated, thanks to bipartite relationships, with another multiplex network containing nodes of different nature. Proceeding this way, a network both multiplex and heterogeneous will be generated. To do so, folow the instructions detailed below</p>
<p>Please note that this version of the package does not deal with directed networks. New features will be included in future updated versions of <em><a href="https://bioconductor.org/packages/3.11/RandomWalkRestartMH">RandomWalkRestartMH</a></em>.</p>
</div>
<div id="installation-of-the-randomwalkrestartmh-package" class="section level1">
<h1 class="hasAnchor">
<a href="#installation-of-the-randomwalkrestartmh-package" class="anchor"></a>Installation of the <em><a href="https://bioconductor.org/packages/3.11/RandomWalkRestartMH">RandomWalkRestartMH</a></em> package</h1>
<p>First of all, you need a current version of <em>R</em>. <em><a href="https://bioconductor.org/packages/3.11/RandomWalkRestartMH">RandomWalkRestartMH</a></em> is a freely available package deposited on <em>Bioconductor</em> and <a href="https://github.com/alberto-valdeolivas/RandomWalkRestartMH/">GitHub</a>. You can install it by running the following commands on an <em>R</em> console:</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r"><span class="kw">if</span> (!<span class="fu"><a href="https://rdrr.io/r/base/ns-load.html">requireNamespace</a></span>(<span class="st">"BiocManager"</span>, <span class="kw">quietly</span> <span class="kw">=</span> <span class="fl">TRUE</span>))
    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html">install.packages</a></span>(<span class="st">"BiocManager"</span>)

<span class="kw pkg">BiocManager</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html">install</a></span>(<span class="st">"RandomWalkRestartMH"</span>)</pre></body></html></div>
<p>or to install the latest version from <a href="https://github.com/alberto-valdeolivas/RandomWalkRestartMH/">GitHub</a> before it is released in Bioconductor:</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r"><span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">"alberto-valdeolivas/RandomWalkRestartMH"</span>)</pre></body></html></div>
</div>
<div id="a-detailed-workflow" class="section level1">
<h1 class="hasAnchor">
<a href="#a-detailed-workflow" class="anchor"></a>A Detailed Workflow</h1>
<p>In the following paragraphs, we describe how to use the <em><a href="https://bioconductor.org/packages/3.11/RandomWalkRestartMH">RandomWalkRestartMH</a></em> package to perform RWR on different types of biological networks. Concretely, we use a protein-protein interaction (PPI) network, a pathway network, a disease-disease similarity network and combinations thereof. These networks are obtained as detailed in <span class="citation">(Valdeolivas et al. 2018)</span>. The PPI and the Pathway network were reduced by only considering genes/proteins expressed in the adipose tissue, in order to reduce the computation time of this vignette.</p>
<p>The goal in the example presented here is, as described in <span class="citation">(Valdeolivas et al. 2018)</span>, to find candidate genes potentially associated with diseases by a guilt-by-association approach. This is based on the fact that genes/proteins with similar functions or similar phenotypes tend to lie closer in biological networks. Therefore, the larger the RWR score of a gene, the more likely it is to be functionally related with the seeds.</p>
<p>We focus on a real biological example: the SHORT syndrome (MIM code: 269880) and its causative gene <em>PIK3R1</em> as described in <span class="citation">(Valdeolivas et al. 2018)</span>. We will see throughout the following paragraphs how the RWR results evolve due to the the integration and exploration of additional networks.</p>
<div id="random-walk-with-restart-on-a-monoplex-network" class="section level2">
<h2 class="hasAnchor">
<a href="#random-walk-with-restart-on-a-monoplex-network" class="anchor"></a>Random Walk with Restart on a Monoplex Network</h2>
<p>RWR has usually been applied within the framework of single PPI networks in bioinformatics <span class="citation">(Kohler et al. 2008)</span>. A gene or a set of genes, so-called seed(s), known to be implicated in a concrete function or in a specific disease, are chosen as the starting point(s) of the algorithm. The RWR particle explores the neighbourhood of the seeds and the algorithm computes a score for all the nodes of the network. The larger it is the score of a node, the closer it is to the seed(s).</p>
<p>Let us generate an object of the class <code>Multiplex</code>, even if it is a monoplex network, with our PPI network.</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">RandomWalkRestartMH</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">igraph</span>)</pre></body></html></div>
<pre><code>## 
## Attaching package: 'igraph'</code></pre>
<pre><code>## The following objects are masked from 'package:stats':
## 
##     decompose, spectrum</code></pre>
<pre><code>## The following object is masked from 'package:base':
## 
##     union</code></pre>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">PPI_Network</span>) <span class="co"># We load the PPI_Network</span>

<span class="co">## We create a Multiplex object composed of 1 layer (It's a Monoplex Network) </span>
<span class="co">## and we display how it looks like</span>
<span class="no">PPI_MultiplexObject</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/create.multiplex.html">create.multiplex</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">PPI</span><span class="kw">=</span><span class="no">PPI_Network</span>))
<span class="no">PPI_MultiplexObject</span></pre></body></html></div>
<pre><code>## Number of Layers:
## [1] 1
## 
## Number of Nodes:
## [1] 4317
## 
## IGRAPH 61e7148 UN-- 4317 18062 -- 
## + attr: name (v/c), type (e/c)
## + edges from 61e7148 (vertex names):
##  [1] AAMP   --VPS52     AAMP   --BHLHE40   AAMP   --GABARAPL2 AAMP   --MAP1LC3B 
##  [5] VPS52  --TXN2      VPS52  --DDX6      VPS52  --MFAP1     VPS52  --PRKAA1   
##  [9] VPS52  --LMO4      VPS52  --STX11     VPS52  --KANK2     VPS52  --PPP1R18  
## [13] VPS52  --TXLNA     VPS52  --KIAA1217  VPS52  --VPS28     VPS52  --ATP6V1D  
## [17] VPS52  --TPM3      VPS52  --KIF5B     VPS52  --NOP2      VPS52  --RNF41    
## [21] VPS52  --WTAP      VPS52  --MAPK3     VPS52  --ZMAT2     VPS52  --VPS51    
## [25] BHLHE40--AES       BHLHE40--PRKAA1    BHLHE40--CCNK      BHLHE40--RBPMS    
## [29] BHLHE40--COX5B     BHLHE40--UBE2I     BHLHE40--MAGED1    BHLHE40--PLEKHB2  
## + ... omitted several edges</code></pre>
<p>To apply the RWR on a monoplex network, we need to compute the adjacency matrix of the network and normalize it by column <span class="citation">(Kohler et al. 2008)</span>, as follows:</p>
<div class="sourceCode" id="cb9"><html><body><pre class="r"><span class="no">AdjMatrix_PPI</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/compute.adjacency.matrix.html">compute.adjacency.matrix</a></span>(<span class="no">PPI_MultiplexObject</span>)
<span class="no">AdjMatrixNorm_PPI</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/normalize.multiplex.adjacency.html">normalize.multiplex.adjacency</a></span>(<span class="no">AdjMatrix_PPI</span>)</pre></body></html></div>
<p>Then, we need to define the seed(s) before running the RWR algorithm on this PPI network. As commented above, we are focusing on the example of the SHORT syndrome. Therefore, we take the <em>PIK3R1</em> gene as seed, and we execute RWR.</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="no">SeedGene</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"PIK3R1"</span>)
<span class="co">## We launch the algorithm with the default parameters (See details on manual)</span>
<span class="no">RWR_PPI_Results</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/Random.Walk.Restart.Multiplex.html">Random.Walk.Restart.Multiplex</a></span>(<span class="no">AdjMatrixNorm_PPI</span>,
                        <span class="no">PPI_MultiplexObject</span>,<span class="no">SeedGene</span>)
<span class="co"># We display the results</span>
<span class="no">RWR_PPI_Results</span></pre></body></html></div>
<pre><code>## Top 10 ranked Nodes:
##    NodeNames       Score
## 1       GRB2 0.006845881
## 2       EGFR 0.006169129
## 3        CRK 0.005674261
## 4       ABL1 0.005617041
## 5        FYN 0.005611086
## 6      CDC42 0.005594680
## 7       SHC1 0.005577900
## 8       CRKL 0.005509182
## 9    KHDRBS1 0.005443541
## 10     TYRO3 0.005441887
## 
## Seed Nodes used:
## [1] "PIK3R1"</code></pre>
<p>Finally, we can create a network (an <code>igraph</code> object) with the top scored genes. Visualize the top results within their interaction network is always a good idea in order to prioritize genes, since we can have a global view of all the potential candidates. The results are presented in Figure 1</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="co">## In this case we selected to induce a network with the Top 15 genes.</span>
<span class="no">TopResults_PPI</span> <span class="kw">&lt;-</span>
    <span class="fu"><a href="../reference/create.multiplexNetwork.topResults.html">create.multiplexNetwork.topResults</a></span>(<span class="no">RWR_PPI_Results</span>,<span class="no">PPI_MultiplexObject</span>,
        <span class="kw">k</span><span class="kw">=</span><span class="fl">15</span>)</pre></body></html></div>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="co">## We print that cluster with its interactions.</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mar</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.1</span>,<span class="fl">0.1</span>,<span class="fl">0.1</span>,<span class="fl">0.1</span>))
<span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">TopResults_PPI</span>, <span class="kw">vertex.label.color</span><span class="kw">=</span><span class="st">"black"</span>,<span class="kw">vertex.frame.color</span><span class="kw">=</span><span class="st">"#ffffff"</span>,
    <span class="kw">vertex.size</span><span class="kw">=</span> <span class="fl">20</span>, <span class="kw">edge.curved</span><span class="kw">=</span><span class="fl">.2</span>,
    <span class="kw">vertex.color</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="kw pkg">igraph</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="no">TopResults_PPI</span>)$<span class="no">name</span> <span class="kw">==</span> <span class="st">"PIK3R1"</span>,<span class="st">"yellow"</span>,
    <span class="st">"#00CCFF"</span>), <span class="kw">edge.color</span><span class="kw">=</span><span class="st">"blue"</span>,<span class="kw">edge.width</span><span class="kw">=</span><span class="fl">0.8</span>)</pre></body></html></div>
<div class="figure">
<img src="RandomWalkRestartMH_files/figure-html/Figure1-1.png" alt="Figure 1: RWR on a monoplex PPI Network. Network representation of the top 15 ranked genes when the RWR algorithm is executed using the PIK3R1 gene as seed (yellow node). Blue edges represent PPI interactions." width="3000"><p class="caption">
Figure 1: RWR on a monoplex PPI Network. Network representation of the top 15 ranked genes when the RWR algorithm is executed using the PIK3R1 gene as seed (yellow node). Blue edges represent PPI interactions.
</p>
</div>
</div>
<div id="random-walk-with-restart-on-a-heterogeneous-network" class="section level2">
<h2 class="hasAnchor">
<a href="#random-walk-with-restart-on-a-heterogeneous-network" class="anchor"></a>Random Walk with Restart on a Heterogeneous Network</h2>
<p>A RWR on a heterogeneous (RWR-H) biological network was described by <span class="citation">(Li and Patra 2010)</span>. They connected a PPI network with a disease-disease similarity network using known gene-disease associations. In this case, genes and/or diseases can be used as seed nodes for the algorithm. In the following example, we also use a heterogeneous network integrating a PPI and a disease-disease similarity network. However, the procedure to obtain these networks is different to the one proposed in <span class="citation">(Li and Patra 2010)</span>, and the details are described in our article <span class="citation">(Valdeolivas et al. 2018)</span>.</p>
<p>To generate a PPI-disease heterogeneous network object, we load the disease-disease network, and combine it with our previously defined <code>Multiplex</code> object containing the PPI network, thanks to the gene-diseases associations obtained from OMIM <span class="citation">(Hamosh et al. 2005)</span>. A <code>MultiplexHet</code> object will be created, even if we are dealing with a monoplex-heterogeneous network.</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">Disease_Network</span>) <span class="co"># We load our disease Network </span>

<span class="co">## We create a multiplex object for the monoplex disease Network</span>
<span class="no">Disease_MultiplexObject</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/create.multiplex.html">create.multiplex</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">Disease</span><span class="kw">=</span><span class="no">Disease_Network</span>))

<span class="co">## We load a data frame containing the gene-disease associations.</span>
<span class="co">## See ?create.multiplexHet for details about its format</span>
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">GeneDiseaseRelations</span>)

<span class="co">## We keep gene-diseases associations where genes are present in the PPI</span>
<span class="co">## network</span>
<span class="no">GeneDiseaseRelations_PPI</span> <span class="kw">&lt;-</span>
    <span class="no">GeneDiseaseRelations</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">GeneDiseaseRelations</span>$<span class="no">hgnc_symbol</span> <span class="kw">%in%</span>
    <span class="no">PPI_MultiplexObject</span>$<span class="no">Pool_of_Nodes</span>),]

<span class="co">## We create the MultiplexHet object.</span>
<span class="no">PPI_Disease_Net</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/create.multiplexHet.html">create.multiplexHet</a></span>(<span class="no">PPI_MultiplexObject</span>,
    <span class="no">Disease_MultiplexObject</span>, <span class="no">GeneDiseaseRelations_PPI</span>)</pre></body></html></div>
<pre><code>## checking input arguments...</code></pre>
<pre><code>## Generating bipartite matrix...</code></pre>
<pre><code>## Expanding bipartite matrix to fit the multiplex network...</code></pre>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="co">## The results look like that</span>
<span class="no">PPI_Disease_Net</span></pre></body></html></div>
<pre><code>## Number of Layers Multiplex 1:
## [1] 1
## 
## Number of Nodes Multiplex 1:
## [1] 4317
## 
## IGRAPH 61e7148 UN-- 4317 18062 -- 
## + attr: name (v/c), type (e/c)
## + edges from 61e7148 (vertex names):
##  [1] AAMP   --VPS52     AAMP   --BHLHE40   AAMP   --GABARAPL2 AAMP   --MAP1LC3B 
##  [5] VPS52  --TXN2      VPS52  --DDX6      VPS52  --MFAP1     VPS52  --PRKAA1   
##  [9] VPS52  --LMO4      VPS52  --STX11     VPS52  --KANK2     VPS52  --PPP1R18  
## [13] VPS52  --TXLNA     VPS52  --KIAA1217  VPS52  --VPS28     VPS52  --ATP6V1D  
## [17] VPS52  --TPM3      VPS52  --KIF5B     VPS52  --NOP2      VPS52  --RNF41    
## [21] VPS52  --WTAP      VPS52  --MAPK3     VPS52  --ZMAT2     VPS52  --VPS51    
## [25] BHLHE40--AES       BHLHE40--PRKAA1    BHLHE40--CCNK      BHLHE40--RBPMS    
## [29] BHLHE40--COX5B     BHLHE40--UBE2I     BHLHE40--MAGED1    BHLHE40--PLEKHB2  
## + ... omitted several edges
## 
## Number of Layers Multiplex 2:
## [1] 1
## 
## Number of Nodes Multiplex 2:
## [1] 6947
## 
## IGRAPH 79f5a3e UN-- 6947 28246 -- 
## + attr: name (v/c), type (e/c)
## + edges from 79f5a3e (vertex names):
##  [1] 100050--122470 100050--227330 100050--259200 100050--305400 100050--601803
##  [6] 100070--105800 100070--105805 100070--107550 100070--120000 100070--130090
## [11] 100070--132900 100070--154750 100070--180300 100070--192310 100070--208060
## [16] 100070--210050 100070--219100 100070--252350 100070--277175 100070--300537
## [21] 100070--309520 100070--600459 100070--604308 100070--606519 100070--608967
## [26] 100070--609192 100070--610168 100070--610380 100070--611788 100070--613780
## [31] 100070--613834 100070--614042 100070--614437 100070--614980 100070--610655
## [36] 100070--615436 100070--616166 100100--192350 100100--236700 100100--236730
## + ... omitted several edges</code></pre>
<p>To apply the RWR-H on a heterogeneous network, we need to compute a matrix that accounts for all the possible transitions of the RWR particle within that network <span class="citation">(Li and Patra 2010)</span>.</p>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="no">PPIHetTranMatrix</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/compute.transition.matrix.html">compute.transition.matrix</a></span>(<span class="no">PPI_Disease_Net</span>)</pre></body></html></div>
<pre><code>## Computing adjacency matrix of the first Multiplex network...</code></pre>
<pre><code>## Computing adjacency matrix of the second Multiplex network...</code></pre>
<pre><code>## Computing inter-subnetworks transitions...</code></pre>
<pre><code>## Computing intra-subnetworks transitions...</code></pre>
<pre><code>## Combining inter e intra layer probabilities into the global 
##           Transition Matix</code></pre>
<p>Before running RWR-H on this PPI-disease heterogeneous network, we need to define the seed(s). As in the previous paragraph, we take <em>PIK3R1</em> as a seed gene. In addition, we can now set the SHORT syndrome itself as a seed disease.</p>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="no">SeedDisease</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"269880"</span>)

<span class="co">## We launch the algorithm with the default parameters (See details on manual)</span>
<span class="no">RWRH_PPI_Disease_Results</span> <span class="kw">&lt;-</span>
    <span class="fu"><a href="../reference/Random.Walk.Restart.MultiplexHet.html">Random.Walk.Restart.MultiplexHet</a></span>(<span class="no">PPIHetTranMatrix</span>,
    <span class="no">PPI_Disease_Net</span>,<span class="no">SeedGene</span>,<span class="no">SeedDisease</span>)

<span class="co"># We display the results</span>
<span class="no">RWRH_PPI_Disease_Results</span></pre></body></html></div>
<pre><code>## Top 10 ranked global nodes:
##    NodeNames       Score
## 1     PIK3R1 0.414259752
## 2     269880 0.372567325
## 3     615214 0.020817435
## 4     616005 0.020785895
## 5     194050 0.005868407
## 6     309000 0.005687206
## 7     262500 0.005681162
## 8     138920 0.005655559
## 9     223370 0.005655117
## 10    608612 0.005649291
## 
## Top 10 ranked nodes from the first Multiplex:
##      NodeNames       Score
## 1479      GRB2 0.001965500
## 1081      EGFR 0.001754048
## 797        CRK 0.001636329
## 603      CDC42 0.001630575
## 19        ABL1 0.001623304
## 798       CRKL 0.001605543
## 1360       FYN 0.001598567
## 3405      SHC1 0.001597720
## 2583    PDGFRB 0.001596830
## 4027     TYRO3 0.001589911
## 
## Top 10 ranked nodes from the second Multiplex:
##      NodeNames       Score
## 6352    615214 0.020817435
## 6705    616005 0.020785895
## 1699    194050 0.005868407
## 3625    309000 0.005687206
## 2901    262500 0.005681162
## 686     138920 0.005655559
## 2150    223370 0.005655117
## 4770    608612 0.005649291
## 4464    606176 0.005641777
## 1411    180500 0.005639919
## 
## Seeds used:
## [1] "PIK3R1" "269880"</code></pre>
<p>Finally, we can create a heterogeneous network (an <code>igraph</code> object) with the top scored genes and the top scored diseases. The results are presented in Figure 2.</p>
<div class="sourceCode" id="cb28"><html><body><pre class="r"><span class="co">## In this case we select to induce a network with the Top 10 genes</span>
<span class="co">## and the Top 10 diseases.</span>
<span class="no">TopResults_PPI_Disease</span> <span class="kw">&lt;-</span>
    <span class="fu"><a href="../reference/create.multiplexHetNetwork.topResults.html">create.multiplexHetNetwork.topResults</a></span>(<span class="no">RWRH_PPI_Disease_Results</span>,
    <span class="no">PPI_Disease_Net</span>, <span class="no">GeneDiseaseRelations_PPI</span>, <span class="kw">k</span><span class="kw">=</span><span class="fl">10</span>)</pre></body></html></div>
<div class="sourceCode" id="cb29"><html><body><pre class="r"><span class="co">## We print that cluster with its interactions.</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mar</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.1</span>,<span class="fl">0.1</span>,<span class="fl">0.1</span>,<span class="fl">0.1</span>))
<span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">TopResults_PPI_Disease</span>, <span class="kw">vertex.label.color</span><span class="kw">=</span><span class="st">"black"</span>,
    <span class="kw">vertex.frame.color</span><span class="kw">=</span><span class="st">"#ffffff"</span>,
    <span class="kw">vertex.size</span><span class="kw">=</span> <span class="fl">20</span>, <span class="kw">edge.curved</span><span class="kw">=</span><span class="fl">.2</span>,
    <span class="kw">vertex.color</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="no">TopResults_PPI_Disease</span>)$<span class="no">name</span> <span class="kw">==</span> <span class="st">"PIK3R1"</span>
    <span class="kw">|</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="no">TopResults_PPI_Disease</span>)$<span class="no">name</span> <span class="kw">==</span> <span class="st">"269880"</span>,<span class="st">"yellow"</span>,
    <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="no">TopResults_PPI_Disease</span>)$<span class="no">name</span> <span class="kw">%in%</span>
        <span class="no">PPI_Disease_Net</span>$<span class="no">Multiplex1</span>$<span class="no">Pool_of_Nodes</span>,<span class="st">"#00CCFF"</span>,<span class="st">"Grey75"</span>)),
    <span class="kw">edge.color</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(<span class="no">TopResults_PPI_Disease</span>)$<span class="no">type</span> <span class="kw">==</span> <span class="st">"PPI"</span>,<span class="st">"blue"</span>,
        <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(<span class="no">TopResults_PPI_Disease</span>)$<span class="no">type</span> <span class="kw">==</span> <span class="st">"Disease"</span>,<span class="st">"black"</span>,<span class="st">"grey50"</span>)),
    <span class="kw">edge.width</span><span class="kw">=</span><span class="fl">0.8</span>,
    <span class="kw">edge.lty</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(<span class="no">TopResults_PPI_Disease</span>)$<span class="no">type</span> <span class="kw">==</span> <span class="st">"bipartiteRelations"</span>,
        <span class="fl">2</span>,<span class="fl">1</span>),
    <span class="kw">vertex.shape</span><span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="no">TopResults_PPI_Disease</span>)$<span class="no">name</span> <span class="kw">%in%</span>
        <span class="no">PPI_Disease_Net</span>$<span class="no">Multiplex1</span>$<span class="no">Pool_of_Nodes</span>,<span class="st">"circle"</span>,<span class="st">"rectangle"</span>))</pre></body></html></div>
<div class="figure">
<img src="RandomWalkRestartMH_files/figure-html/Figure2-1.png" alt="Figure 2: RWR-H on a heterogeneous PPI-Disease Network. Network representation of the top 10 ranked genes and the top 10 ranked diseases when the RWR-H algorithm is executed using the PIK3R1 gene and the SHORT syndrome disease (MIM code: 269880) as seeds (yellow nodes). Circular nodes represent genes and rectangular nodes show diseases. Blue edges are PPI interactions and black edges are similarity links between diseases. Dashed edges are the bipartite gene-disease associations." width="3000"><p class="caption">
Figure 2: RWR-H on a heterogeneous PPI-Disease Network. Network representation of the top 10 ranked genes and the top 10 ranked diseases when the RWR-H algorithm is executed using the PIK3R1 gene and the SHORT syndrome disease (MIM code: 269880) as seeds (yellow nodes). Circular nodes represent genes and rectangular nodes show diseases. Blue edges are PPI interactions and black edges are similarity links between diseases. Dashed edges are the bipartite gene-disease associations.
</p>
</div>
</div>
<div id="random-walk-with-restart-on-a-multiplex-network" class="section level2">
<h2 class="hasAnchor">
<a href="#random-walk-with-restart-on-a-multiplex-network" class="anchor"></a>Random Walk with Restart on a Multiplex Network</h2>
<p>Some limitations can arise when single networks are used to represent and describe systems whose entities can interact through more than one type of connections <span class="citation">(Battiston, Nicosia, and Latora 2014)</span>. This is the case of social interactions, transportation networks or biological systems, among others. The Multiplex framework provides an appealing approach to describe these systems, since they are able to integrate this diversity of data while keeping track of the original features and topologies of the different sources.</p>
<p>Consequently, algorithms able to exploit the information stored on multiplex networks should improve the results provided by methods operating on single networks. In this context, we extended the random walk with restart algorithm to multiplex networks (RWR-M) <span class="citation">(Valdeolivas et al. 2018)</span>.</p>
<p>In the following example, we create a multiplex network integrated by our PPI network and a network derived from pathway databases <span class="citation">(Valdeolivas et al. 2018)</span>.</p>
<div class="sourceCode" id="cb30"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">Pathway_Network</span>) <span class="co"># We load the Pathway Network</span>

<span class="co">## We create a 2-layers Multiplex object</span>
<span class="no">PPI_PATH_Multiplex</span> <span class="kw">&lt;-</span>
  <span class="fu"><a href="../reference/create.multiplex.html">create.multiplex</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">PPI</span><span class="kw">=</span><span class="no">PPI_Network</span>,<span class="kw">PATH</span><span class="kw">=</span><span class="no">Pathway_Network</span>))
<span class="no">PPI_PATH_Multiplex</span></pre></body></html></div>
<pre><code>## Number of Layers:
## [1] 2
## 
## Number of Nodes:
## [1] 4899
## 
## IGRAPH ae89d80 UN-- 4899 18062 -- 
## + attr: name (v/c), type (e/c)
## + edges from ae89d80 (vertex names):
##  [1] AAMP   --VPS52     AAMP   --BHLHE40   AAMP   --GABARAPL2 AAMP   --MAP1LC3B 
##  [5] VPS52  --TXN2      VPS52  --DDX6      VPS52  --MFAP1     VPS52  --PRKAA1   
##  [9] VPS52  --LMO4      VPS52  --STX11     VPS52  --KANK2     VPS52  --PPP1R18  
## [13] VPS52  --TXLNA     VPS52  --KIAA1217  VPS52  --VPS28     VPS52  --ATP6V1D  
## [17] VPS52  --TPM3      VPS52  --KIF5B     VPS52  --NOP2      VPS52  --RNF41    
## [21] VPS52  --WTAP      VPS52  --MAPK3     VPS52  --ZMAT2     VPS52  --VPS51    
## [25] BHLHE40--AES       BHLHE40--PRKAA1    BHLHE40--CCNK      BHLHE40--RBPMS    
## [29] BHLHE40--COX5B     BHLHE40--UBE2I     BHLHE40--MAGED1    BHLHE40--PLEKHB2  
## + ... omitted several edges
## 
## IGRAPH b47b9bb UN-- 4899 62602 -- 
## + attr: name (v/c), type (e/c)
## + edges from b47b9bb (vertex names):
##  [1] BANF1--PSIP1   BANF1--HMGA1   BANF1--PPP2R1A BANF1--PPP2CA  BANF1--KPNA1  
##  [6] BANF1--TPR     BANF1--NUP62   BANF1--NUP153  BANF1--RANBP2  BANF1--NUP54  
## [11] BANF1--POM121  BANF1--NUP85   BANF1--PPP2R2A BANF1--EMD     BANF1--LEMD2  
## [16] BANF1--ANKLE2  PSIP1--HMGA1   PSIP1--KPNA1   PSIP1--TPR     PSIP1--NUP62  
## [21] PSIP1--NUP153  PSIP1--RANBP2  PSIP1--NUP54   PSIP1--POM121  PSIP1--NUP85  
## [26] HMGA1--TP53    HMGA1--KPNA1   HMGA1--TPR     HMGA1--NUP62   HMGA1--NUP153 
## [31] HMGA1--RANBP2  HMGA1--NUP54   HMGA1--POM121  HMGA1--NUP85   HMGA1--MYC    
## [36] HMGA1--RB1     HMGA1--MAX     HMGA1--RPS6KB1 XRCC6--XRCC5   XRCC6--IRF3   
## + ... omitted several edges</code></pre>
<p>Afterwards, as in the monoplex case, we have to compute and normalize the adjacency matrix of the multiplex network.</p>
<div class="sourceCode" id="cb32"><html><body><pre class="r"><span class="no">AdjMatrix_PPI_PATH</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/compute.adjacency.matrix.html">compute.adjacency.matrix</a></span>(<span class="no">PPI_PATH_Multiplex</span>)
<span class="no">AdjMatrixNorm_PPI_PATH</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/normalize.multiplex.adjacency.html">normalize.multiplex.adjacency</a></span>(<span class="no">AdjMatrix_PPI_PATH</span>)</pre></body></html></div>
<p>Then, we set again as seed the <em>PIK3R1</em> gene and we perform RWR-M on this new multiplex network.</p>
<div class="sourceCode" id="cb33"><html><body><pre class="r"><span class="co">## We launch the algorithm with the default parameters (See details on manual)</span>
<span class="no">RWR_PPI_PATH_Results</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/Random.Walk.Restart.Multiplex.html">Random.Walk.Restart.Multiplex</a></span>(<span class="no">AdjMatrixNorm_PPI_PATH</span>,
                        <span class="no">PPI_PATH_Multiplex</span>,<span class="no">SeedGene</span>)
<span class="co"># We display the results</span>
<span class="no">RWR_PPI_PATH_Results</span></pre></body></html></div>
<pre><code>## Top 10 ranked Nodes:
##    NodeNames       Score
## 1       GRB2 0.001662893
## 2        FYN 0.001517786
## 3       HCST 0.001506594
## 4       EGFR 0.001494459
## 5       SHC1 0.001492456
## 6       PTK2 0.001430282
## 7       JAK2 0.001401867
## 8       HRAS 0.001396854
## 9       CRKL 0.001389391
## 10    PDGFRB 0.001369533
## 
## Seed Nodes used:
## [1] "PIK3R1"</code></pre>
<p>Finally, we can create a multiplex network (an <code>igraph</code> object) with the top scored genes. The results are presented in Figure 3.</p>
<div class="sourceCode" id="cb35"><html><body><pre class="r"><span class="co">## In this case we select to induce a multiplex network with the Top 15 genes.</span>
<span class="no">TopResults_PPI_PATH</span> <span class="kw">&lt;-</span>
    <span class="fu"><a href="../reference/create.multiplexNetwork.topResults.html">create.multiplexNetwork.topResults</a></span>(<span class="no">RWR_PPI_PATH_Results</span>, <span class="no">PPI_PATH_Multiplex</span>,
      <span class="kw">k</span><span class="kw">=</span><span class="fl">15</span>)</pre></body></html></div>
<div class="sourceCode" id="cb36"><html><body><pre class="r"><span class="co">## We print that cluster with its interactions.</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mar</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.1</span>,<span class="fl">0.1</span>,<span class="fl">0.1</span>,<span class="fl">0.1</span>))
<span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">TopResults_PPI_PATH</span>, <span class="kw">vertex.label.color</span><span class="kw">=</span><span class="st">"black"</span>,
    <span class="kw">vertex.frame.color</span><span class="kw">=</span><span class="st">"#ffffff"</span>, <span class="kw">vertex.size</span><span class="kw">=</span> <span class="fl">20</span>,
    <span class="kw">edge.curved</span><span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(<span class="no">TopResults_PPI_PATH</span>)$<span class="no">type</span> <span class="kw">==</span> <span class="st">"PPI"</span>,
                    <span class="fl">0.4</span>,<span class="fl">0</span>),
    <span class="kw">vertex.color</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="kw pkg">igraph</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="no">TopResults_PPI_PATH</span>)$<span class="no">name</span> <span class="kw">==</span> <span class="st">"PIK3R1"</span>,
                    <span class="st">"yellow"</span>,<span class="st">"#00CCFF"</span>),<span class="kw">edge.width</span><span class="kw">=</span><span class="fl">0.8</span>,
    <span class="kw">edge.color</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(<span class="no">TopResults_PPI_PATH</span>)$<span class="no">type</span> <span class="kw">==</span> <span class="st">"PPI"</span>,
                      <span class="st">"blue"</span>,<span class="st">"red"</span>))</pre></body></html></div>
<div class="figure">
<img src="RandomWalkRestartMH_files/figure-html/Figure%203-1.png" alt="Figure 3: RWR-M on a multiplex PPI-Pathway Network. Network representation of the top 15 ranked genes when the RWR-M algorithm is executed using the *PIK3R1* gene (yellownode). Blue curved edges are PPI interactions and red straight edges are Pathways links. All the interactions are aggregated into a monoplex network only for visualization purposes." width="3000"><p class="caption">
Figure 3: RWR-M on a multiplex PPI-Pathway Network. Network representation of the top 15 ranked genes when the RWR-M algorithm is executed using the <em>PIK3R1</em> gene (yellownode). Blue curved edges are PPI interactions and red straight edges are Pathways links. All the interactions are aggregated into a monoplex network only for visualization purposes.
</p>
</div>
</div>
<div id="random-walk-with-restart-on-a-multiplex-heterogeneous-network" class="section level2">
<h2 class="hasAnchor">
<a href="#random-walk-with-restart-on-a-multiplex-heterogeneous-network" class="anchor"></a>Random Walk with Restart on a Multiplex-Heterogeneous Network</h2>
<p>RWR-H and RWR-M remarkably improve the results obtained by classical RWR on monoplex networks, as we demonstrated in the particular case of retrieving known gene-disease associations <span class="citation">(Valdeolivas et al. 2018)</span>. Therefore, an algorithm able to execute a random walk with restart on both, multiplex and heterogeneous networks, is expected to achieve an even better performance. We extended our RWR-M approach to heterogeneous networks, defining a random walk with restart on multiplex-heterogeneous networks (RWR-MH) <span class="citation">(Valdeolivas et al. 2018)</span>.</p>
<p>Let us integrate all the networks described previously (PPI, Pathways and disease-disease similarity) into a multiplex and heterogeneous network. To do so, we connect genes in both multiplex layers (PPI and Pathways) to the disease network, if a bipartite gene-disease relation exists.</p>
<div class="sourceCode" id="cb37"><html><body><pre class="r"><span class="co">## We keep gene-diseases associations where genes are present in the PPI</span>
<span class="co">## or in the pathway network</span>
<span class="no">GeneDiseaseRelations_PPI_PATH</span> <span class="kw">&lt;-</span>
    <span class="no">GeneDiseaseRelations</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">GeneDiseaseRelations</span>$<span class="no">hgnc_symbol</span> <span class="kw">%in%</span>
    <span class="no">PPI_PATH_Multiplex</span>$<span class="no">Pool_of_Nodes</span>),]

<span class="co">## We create the MultiplexHet object.</span>
<span class="no">PPI_PATH_Disease_Net</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/create.multiplexHet.html">create.multiplexHet</a></span>(<span class="no">PPI_PATH_Multiplex</span>,
    <span class="no">Disease_MultiplexObject</span>, <span class="no">GeneDiseaseRelations_PPI_PATH</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Disease"</span>))</pre></body></html></div>
<pre><code>## checking input arguments...</code></pre>
<pre><code>## Generating bipartite matrix...</code></pre>
<pre><code>## Expanding bipartite matrix to fit the multiplex network...</code></pre>
<div class="sourceCode" id="cb41"><html><body><pre class="r"><span class="co">## The results look like that</span>
<span class="no">PPI_PATH_Disease_Net</span></pre></body></html></div>
<pre><code>## Number of Layers Multiplex 1:
## [1] 2
## 
## Number of Nodes Multiplex 1:
## [1] 4899
## 
## IGRAPH ae89d80 UN-- 4899 18062 -- 
## + attr: name (v/c), type (e/c)
## + edges from ae89d80 (vertex names):
##  [1] AAMP   --VPS52     AAMP   --BHLHE40   AAMP   --GABARAPL2 AAMP   --MAP1LC3B 
##  [5] VPS52  --TXN2      VPS52  --DDX6      VPS52  --MFAP1     VPS52  --PRKAA1   
##  [9] VPS52  --LMO4      VPS52  --STX11     VPS52  --KANK2     VPS52  --PPP1R18  
## [13] VPS52  --TXLNA     VPS52  --KIAA1217  VPS52  --VPS28     VPS52  --ATP6V1D  
## [17] VPS52  --TPM3      VPS52  --KIF5B     VPS52  --NOP2      VPS52  --RNF41    
## [21] VPS52  --WTAP      VPS52  --MAPK3     VPS52  --ZMAT2     VPS52  --VPS51    
## [25] BHLHE40--AES       BHLHE40--PRKAA1    BHLHE40--CCNK      BHLHE40--RBPMS    
## [29] BHLHE40--COX5B     BHLHE40--UBE2I     BHLHE40--MAGED1    BHLHE40--PLEKHB2  
## + ... omitted several edges
## 
## IGRAPH b47b9bb UN-- 4899 62602 -- 
## + attr: name (v/c), type (e/c)
## + edges from b47b9bb (vertex names):
##  [1] BANF1--PSIP1   BANF1--HMGA1   BANF1--PPP2R1A BANF1--PPP2CA  BANF1--KPNA1  
##  [6] BANF1--TPR     BANF1--NUP62   BANF1--NUP153  BANF1--RANBP2  BANF1--NUP54  
## [11] BANF1--POM121  BANF1--NUP85   BANF1--PPP2R2A BANF1--EMD     BANF1--LEMD2  
## [16] BANF1--ANKLE2  PSIP1--HMGA1   PSIP1--KPNA1   PSIP1--TPR     PSIP1--NUP62  
## [21] PSIP1--NUP153  PSIP1--RANBP2  PSIP1--NUP54   PSIP1--POM121  PSIP1--NUP85  
## [26] HMGA1--TP53    HMGA1--KPNA1   HMGA1--TPR     HMGA1--NUP62   HMGA1--NUP153 
## [31] HMGA1--RANBP2  HMGA1--NUP54   HMGA1--POM121  HMGA1--NUP85   HMGA1--MYC    
## [36] HMGA1--RB1     HMGA1--MAX     HMGA1--RPS6KB1 XRCC6--XRCC5   XRCC6--IRF3   
## + ... omitted several edges
## 
## Number of Layers Multiplex 2:
## [1] 1
## 
## Number of Nodes Multiplex 2:
## [1] 6947
## 
## IGRAPH 79f5a3e UN-- 6947 28246 -- 
## + attr: name (v/c), type (e/c)
## + edges from 79f5a3e (vertex names):
##  [1] 100050--122470 100050--227330 100050--259200 100050--305400 100050--601803
##  [6] 100070--105800 100070--105805 100070--107550 100070--120000 100070--130090
## [11] 100070--132900 100070--154750 100070--180300 100070--192310 100070--208060
## [16] 100070--210050 100070--219100 100070--252350 100070--277175 100070--300537
## [21] 100070--309520 100070--600459 100070--604308 100070--606519 100070--608967
## [26] 100070--609192 100070--610168 100070--610380 100070--611788 100070--613780
## [31] 100070--613834 100070--614042 100070--614437 100070--614980 100070--610655
## [36] 100070--615436 100070--616166 100100--192350 100100--236700 100100--236730
## + ... omitted several edges</code></pre>
<p>To apply the RWR-MH on a multiplex and heterogeneous network, we need to compute a matrix that accounts for all the possible transitions of the RWR particle within this network <span class="citation">(Valdeolivas et al. 2018)</span>.</p>
<div class="sourceCode" id="cb43"><html><body><pre class="r"><span class="no">PPI_PATH_HetTranMatrix</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/compute.transition.matrix.html">compute.transition.matrix</a></span>(<span class="no">PPI_PATH_Disease_Net</span>)</pre></body></html></div>
<pre><code>## Computing adjacency matrix of the first Multiplex network...</code></pre>
<pre><code>## Computing adjacency matrix of the second Multiplex network...</code></pre>
<pre><code>## Computing inter-subnetworks transitions...</code></pre>
<pre><code>## Computing intra-subnetworks transitions...</code></pre>
<pre><code>## Combining inter e intra layer probabilities into the global 
##           Transition Matix</code></pre>
<p>As in the RWR-H situation, we can take as seeds both, the <em>PIK3R1</em> gene and the the SHORT syndrome disease.</p>
<div class="sourceCode" id="cb49"><html><body><pre class="r"><span class="co">## We launch the algorithm with the default parameters (See details on manual)</span>
<span class="no">RWRMH_PPI_PATH_Disease_Results</span> <span class="kw">&lt;-</span>
    <span class="fu"><a href="../reference/Random.Walk.Restart.MultiplexHet.html">Random.Walk.Restart.MultiplexHet</a></span>(<span class="no">PPI_PATH_HetTranMatrix</span>,
    <span class="no">PPI_PATH_Disease_Net</span>,<span class="no">SeedGene</span>,<span class="no">SeedDisease</span>)

<span class="co"># We display the results</span>
<span class="no">RWRMH_PPI_PATH_Disease_Results</span></pre></body></html></div>
<pre><code>## Top 10 ranked global nodes:
##    NodeNames       Score
## 1     269880 0.372462393
## 2     PIK3R1 0.207021304
## 3     615214 0.020797695
## 4     616005 0.020775842
## 5     194050 0.005868202
## 6     608612 0.005691321
## 7     262500 0.005691027
## 8     309000 0.005683228
## 9     138920 0.005653962
## 10    223370 0.005653358
## 
## Top 10 ranked nodes from the first Multiplex:
##      NodeNames        Score
## 1616       GHR 0.0005153889
## 4867  ZMPSTE24 0.0004781314
## 1706      GRB2 0.0004642946
## 1625      GJA1 0.0004457777
## 1781      HCST 0.0004408101
## 1249      EGFR 0.0004239882
## 1564       FYN 0.0004193109
## 3883      SHC1 0.0004161564
## 2990    PDGFRB 0.0004079884
## 3361      PTK2 0.0004069179
## 
## Top 10 ranked nodes from the second Multiplex:
##      NodeNames       Score
## 6352    615214 0.020797695
## 6705    616005 0.020775842
## 1699    194050 0.005868202
## 4770    608612 0.005691321
## 2901    262500 0.005691027
## 3625    309000 0.005683228
## 686     138920 0.005653962
## 2150    223370 0.005653358
## 4464    606176 0.005639004
## 1411    180500 0.005631733
## 
## Seeds used:
## [1] "PIK3R1" "269880"</code></pre>
<p>Finally, we can create a multiplex and heterogeneous network (an <code>igraph</code> object) with the top scored genes and the top scored diseases. The results are presented in Figure 4.</p>
<div class="sourceCode" id="cb51"><html><body><pre class="r"><span class="co">## In this case we select to induce a network with the Top 10 genes.</span>
<span class="co">## and the Top 10 diseases.</span>
<span class="no">TopResults_PPI_PATH_Disease</span> <span class="kw">&lt;-</span>
    <span class="fu"><a href="../reference/create.multiplexHetNetwork.topResults.html">create.multiplexHetNetwork.topResults</a></span>(<span class="no">RWRMH_PPI_PATH_Disease_Results</span>,
    <span class="no">PPI_PATH_Disease_Net</span>, <span class="no">GeneDiseaseRelations_PPI_PATH</span>, <span class="kw">k</span><span class="kw">=</span><span class="fl">10</span>)</pre></body></html></div>
<div class="sourceCode" id="cb52"><html><body><pre class="r"><span class="co">## We print that cluster with its interactions.</span>
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mar</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.1</span>,<span class="fl">0.1</span>,<span class="fl">0.1</span>,<span class="fl">0.1</span>))
<span class="fu"><a href="https://rdrr.io/r/base/plot.html">plot</a></span>(<span class="no">TopResults_PPI_PATH_Disease</span>, <span class="kw">vertex.label.color</span><span class="kw">=</span><span class="st">"black"</span>,
    <span class="kw">vertex.frame.color</span><span class="kw">=</span><span class="st">"#ffffff"</span>,
    <span class="kw">vertex.size</span><span class="kw">=</span> <span class="fl">20</span>,
    <span class="kw">edge.curved</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(<span class="no">TopResults_PPI_PATH_Disease</span>)$<span class="no">type</span> <span class="kw">==</span> <span class="st">"PATH"</span>,
                    <span class="fl">0</span>,<span class="fl">0.3</span>),
    <span class="kw">vertex.color</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="no">TopResults_PPI_PATH_Disease</span>)$<span class="no">name</span> <span class="kw">==</span> <span class="st">"PIK3R1"</span>
    <span class="kw">|</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="no">TopResults_PPI_Disease</span>)$<span class="no">name</span> <span class="kw">==</span> <span class="st">"269880"</span>,<span class="st">"yellow"</span>,
    <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="no">TopResults_PPI_PATH_Disease</span>)$<span class="no">name</span> <span class="kw">%in%</span>
               <span class="no">PPI_PATH_Disease_Net</span>$<span class="no">Multiplex1</span>$<span class="no">Pool_of_Nodes</span>,
    <span class="st">"#00CCFF"</span>,<span class="st">"Grey75"</span>)),
    <span class="kw">edge.color</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(<span class="no">TopResults_PPI_PATH_Disease</span>)$<span class="no">type</span> <span class="kw">==</span> <span class="st">"PPI"</span>,<span class="st">"blue"</span>,
    <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(<span class="no">TopResults_PPI_PATH_Disease</span>)$<span class="no">type</span> <span class="kw">==</span> <span class="st">"PATH"</span>,<span class="st">"red"</span>,
    <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(<span class="no">TopResults_PPI_PATH_Disease</span>)$<span class="no">type</span> <span class="kw">==</span> <span class="st">"Disease"</span>,<span class="st">"black"</span>,<span class="st">"grey50"</span>))),
    <span class="kw">edge.width</span><span class="kw">=</span><span class="fl">0.8</span>,
    <span class="kw">edge.lty</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(<span class="no">TopResults_PPI_PATH_Disease</span>)$<span class="no">type</span> <span class="kw">==</span>
        <span class="st">"bipartiteRelations"</span>, <span class="fl">2</span>,<span class="fl">1</span>),
    <span class="kw">vertex.shape</span><span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="no">TopResults_PPI_PATH_Disease</span>)$<span class="no">name</span> <span class="kw">%in%</span>
        <span class="no">PPI_PATH_Disease_Net</span>$<span class="no">Multiplex1</span>$<span class="no">Pool_of_Nodes</span>,<span class="st">"circle"</span>,<span class="st">"rectangle"</span>))</pre></body></html></div>
<div class="figure">
<img src="RandomWalkRestartMH_files/figure-html/Figure%204-1.png" alt="Figure 4: RWR-MH on a multiplex and heterogeneous network (PPI-Pathway-Disease). Network representation of the top 10 ranked genes and the top 10 ranked diseases when the RWR-H algorithm is executed using the PIK3R1 gene and the SHORT syndrome disease (MIM code: 269880) as seeds (yellow nodes). Circular nodes represent genes and rectangular nodes show diseases. Blue curved edges are PPI interactions and red straight edges are Pathways links. Black edges are similarity links between diseases. Dashed edges are the bipartite gene-disease associations. Multiplex interactions are aggregated into a monoplex network only for visualization purposes." width="3000"><p class="caption">
Figure 4: RWR-MH on a multiplex and heterogeneous network (PPI-Pathway-Disease). Network representation of the top 10 ranked genes and the top 10 ranked diseases when the RWR-H algorithm is executed using the PIK3R1 gene and the SHORT syndrome disease (MIM code: 269880) as seeds (yellow nodes). Circular nodes represent genes and rectangular nodes show diseases. Blue curved edges are PPI interactions and red straight edges are Pathways links. Black edges are similarity links between diseases. Dashed edges are the bipartite gene-disease associations. Multiplex interactions are aggregated into a monoplex network only for visualization purposes.
</p>
</div>
</div>
<div id="random-walk-with-restart-on-a-full-multiplex-heterogeneous-weighted-network" class="section level2">
<h2 class="hasAnchor">
<a href="#random-walk-with-restart-on-a-full-multiplex-heterogeneous-weighted-network" class="anchor"></a>Random Walk with Restart on a full Multiplex-Heterogeneous weighted Network</h2>
<p>In this section, we do an example of Random Walk with restart on full Multiplex-Heterogeneous network. In addition, we are going to show how to work with weighted networks. Indeed, one just need to include a weight attribute in the igraph objects. The user can also weight the bipartite relations by including a third column in the data frame with the weights.</p>
<div class="sourceCode" id="cb53"><html><body><pre class="r"><span class="co">## I first include aleatory weights in the previously used networks</span>
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="fl">124</span>)
<span class="no">PPI_Network</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/set_edge_attr.html">set_edge_attr</a></span>(<span class="no">PPI_Network</span>,<span class="st">"weight"</span>,<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(<span class="no">PPI_Network</span>),
  <span class="kw">value</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/gsize.html">ecount</a></span>(<span class="no">PPI_Network</span>)))
<span class="no">Pathway_Network</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/set_edge_attr.html">set_edge_attr</a></span>(<span class="no">Pathway_Network</span>,<span class="st">"weight"</span>,<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(<span class="no">Pathway_Network</span>),
  <span class="kw">value</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/gsize.html">ecount</a></span>(<span class="no">Pathway_Network</span>)))
<span class="no">Disease_Network_1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/set_edge_attr.html">set_edge_attr</a></span>(<span class="no">Disease_Network</span>,<span class="st">"weight"</span>,<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/E.html">E</a></span>(<span class="no">Disease_Network</span>),
  <span class="kw">value</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html">runif</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/igraph/man/gsize.html">ecount</a></span>(<span class="no">Disease_Network</span>)))

<span class="co">## I am also going to generate a second layer for the disease network </span>
<span class="co">## from random combinations of elements from the disease network (edges)</span>
<span class="no">allNames</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html">V</a></span>(<span class="no">Disease_Network</span>)$<span class="no">name</span>
<span class="no">vectorNames</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/t.html">t</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/combn.html">combn</a></span>(<span class="no">allNames</span>,<span class="fl">2</span>))
<span class="no">idx</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(<span class="no">vectorNames</span>)),<span class="kw">size</span><span class="kw">=</span> <span class="fl">10000</span>)

<span class="no">Disease_Network_2</span> <span class="kw">&lt;-</span>
  <span class="fu"><a href="https://rdrr.io/pkg/igraph/man/graph_from_data_frame.html">graph_from_data_frame</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="no">vectorNames</span>[<span class="no">idx</span>,]), <span class="kw">directed</span> <span class="kw">=</span> <span class="fl">FALSE</span>)

<span class="co">## We create the multiplex objects and multiplex heterogeneous objects as </span>
<span class="co">## usually</span>

<span class="no">PPI_PATH_Multiplex</span> <span class="kw">&lt;-</span>
    <span class="fu"><a href="../reference/create.multiplex.html">create.multiplex</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">PPI</span><span class="kw">=</span><span class="no">PPI_Network</span>, <span class="kw">PATH</span><span class="kw">=</span><span class="no">Pathway_Network</span>))
<span class="no">Disease_MultiplexObject</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/create.multiplex.html">create.multiplex</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="kw">Disease1</span><span class="kw">=</span><span class="no">Disease_Network_1</span>,
    <span class="kw">Disease2</span> <span class="kw">=</span> <span class="no">Disease_Network_2</span>))

<span class="no">GeneDiseaseRelations_PPI_PATH</span> <span class="kw">&lt;-</span>
    <span class="no">GeneDiseaseRelations</span>[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">GeneDiseaseRelations</span>$<span class="no">hgnc_symbol</span> <span class="kw">%in%</span>
                               <span class="no">PPI_PATH_Multiplex</span>$<span class="no">Pool_of_Nodes</span>),]

<span class="no">PPI_PATH_Disease_Net</span> <span class="kw">&lt;-</span>
  <span class="fu"><a href="../reference/create.multiplexHet.html">create.multiplexHet</a></span>(<span class="no">PPI_PATH_Multiplex</span>,<span class="no">Disease_MultiplexObject</span>,
    <span class="no">GeneDiseaseRelations_PPI_PATH</span>)</pre></body></html></div>
<pre><code>## checking input arguments...</code></pre>
<pre><code>## Generating bipartite matrix...</code></pre>
<pre><code>## Expanding bipartite matrix to fit the multiplex network...</code></pre>
<div class="sourceCode" id="cb57"><html><body><pre class="r"><span class="no">PPI_PATH_HetTranMatrix</span> <span class="kw">&lt;-</span> <span class="fu"><a href="../reference/compute.transition.matrix.html">compute.transition.matrix</a></span>(<span class="no">PPI_PATH_Disease_Net</span>)</pre></body></html></div>
<pre><code>## Computing adjacency matrix of the first Multiplex network...</code></pre>
<pre><code>## Computing adjacency matrix of the second Multiplex network...</code></pre>
<pre><code>## Computing inter-subnetworks transitions...</code></pre>
<pre><code>## Computing intra-subnetworks transitions...</code></pre>
<pre><code>## Combining inter e intra layer probabilities into the global 
##           Transition Matix</code></pre>
<div class="sourceCode" id="cb63"><html><body><pre class="r"><span class="no">SeedDisease</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"269880"</span>)
<span class="no">SeedGene</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"PIK3R1"</span>)

<span class="no">RWRH_PPI_PATH_Disease_Results</span> <span class="kw">&lt;-</span><span class="fu"><a href="../reference/Random.Walk.Restart.MultiplexHet.html">Random.Walk.Restart.MultiplexHet</a></span>(<span class="no">PPI_PATH_HetTranMatrix</span>,
    <span class="no">PPI_PATH_Disease_Net</span>,<span class="no">SeedGene</span>,<span class="no">SeedDisease</span>)</pre></body></html></div>
<p>We can see that the results have changed due to the weights and the additional layer in the disease multiplex network.</p>
<div class="sourceCode" id="cb64"><html><body><pre class="r"><span class="no">RWRH_PPI_PATH_Disease_Results</span></pre></body></html></div>
<pre><code>## Top 10 ranked global nodes:
##    NodeNames        Score
## 1     PIK3R1 0.2087238755
## 2     269880 0.1931551706
## 3     616005 0.0110511284
## 4     615214 0.0108458033
## 5     310400 0.0026568065
## 6     616095 0.0017902062
## 7     606176 0.0009476716
## 8    SLC16A1 0.0008403851
## 9     138920 0.0008195664
## 10    608624 0.0007370942
## 
## Top 10 ranked nodes from the first Multiplex:
##      NodeNames        Score
## 3913   SLC16A1 0.0008403851
## 1564       FYN 0.0006876717
## 4160     STAT3 0.0006707869
## 406        AXL 0.0006192449
## 2181      LCP2 0.0005221790
## 3883      SHC1 0.0005103967
## 2745    NFKBIA 0.0004403848
## 1035    DAB2IP 0.0004387133
## 694      CDC42 0.0004311616
## 3361      PTK2 0.0004160867
## 
## Top 10 ranked nodes from the second Multiplex:
##      NodeNames        Score
## 6705    616005 0.0110511284
## 6352    615214 0.0108458033
## 3666    310400 0.0026568065
## 6742    616095 0.0017902062
## 4464    606176 0.0009476716
## 686     138920 0.0008195664
## 4773    608624 0.0007370942
## 3625    309000 0.0007233768
## 1411    180500 0.0005576246
## 4770    608612 0.0005309275
## 
## Seeds used:
## [1] "PIK3R1" "269880"</code></pre>
</div>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session info</h1>
<pre><code>## R version 4.0.2 (2020-06-22)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.1 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
## 
## locale:
##  [1] LC_CTYPE=en_GB.UTF-8       LC_NUMERIC=C              
##  [3] LC_TIME=en_GB.UTF-8        LC_COLLATE=en_GB.UTF-8    
##  [5] LC_MONETARY=en_GB.UTF-8    LC_MESSAGES=en_GB.UTF-8   
##  [7] LC_PAPER=en_GB.UTF-8       LC_NAME=C                 
##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
## [11] LC_MEASUREMENT=en_GB.UTF-8 LC_IDENTIFICATION=C       
## 
## attached base packages:
## [1] stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
## [1] igraph_1.2.5              RandomWalkRestartMH_1.9.1
## [3] BiocStyle_2.16.0         
## 
## loaded via a namespace (and not attached):
##  [1] Rcpp_1.0.5          highr_0.8           compiler_4.0.2     
##  [4] BiocManager_1.30.10 tools_4.0.2         digest_0.6.25      
##  [7] dnet_1.1.7          evaluate_0.14       memoise_1.1.0      
## [10] nlme_3.1-147        lattice_0.20-41     pkgconfig_2.0.3    
## [13] rlang_0.4.7         Matrix_1.2-18       graph_1.66.0       
## [16] rstudioapi_0.11     Rgraphviz_2.32.0    yaml_2.2.1         
## [19] parallel_4.0.2      hexbin_1.28.1       pkgdown_1.5.1      
## [22] xfun_0.16           stringr_1.4.0       knitr_1.29         
## [25] desc_1.2.0          fs_1.4.2            stats4_4.0.2       
## [28] rprojroot_1.3-2     grid_4.0.2          R6_2.4.1           
## [31] rmarkdown_2.3       bookdown_0.20       magrittr_1.5       
## [34] backports_1.1.8     htmltools_0.5.0     MASS_7.3-51.6      
## [37] BiocGenerics_0.34.0 assertthat_0.2.1    ape_5.4            
## [40] supraHex_1.26.0     stringi_1.4.6       crayon_1.3.4</code></pre>
</div>
<div id="references" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#references" class="anchor"></a>References</h1>
<div id="refs" class="references">
<div id="ref-Battiston2014">
<p>Battiston, Federico, Vincenzo Nicosia, and Vito Latora. 2014. â€œStructural measures for multiplex networks.â€ <em>Physical Review E - Statistical, Nonlinear, and Soft Matter Physics</em> 89 (3): 1â€“16. <a href="https://doi.org/10.1103/PhysRevE.89.032804">https://doi.org/10.1103/PhysRevE.89.032804</a>.</p>
</div>
<div id="ref-Hamosh2005">
<p>Hamosh, Ada, Alan F. Scott, Joanna S. Amberger, Carol A. Bocchini, and Victor A. McKusick. 2005. â€œOnline Mendelian Inheritance in Man (OMIM), a knowledgebase of human genes and genetic disorders.â€ <em>Nucleic Acids Research</em> 33 (DATABASE ISS.): 514â€“17. <a href="https://doi.org/10.1093/nar/gki033">https://doi.org/10.1093/nar/gki033</a>.</p>
</div>
<div id="ref-Kohler2008">
<p>Kohler, Sebastian, Sebastian Bauer, Denise Horn, and Peter N Robinson. 2008. â€œAJHG - Walking the Interactome for Prioritization of Candidate Disease Genes,â€ no. April: 949â€“58. <a href="https://doi.org/10.1016/j.ajhg.2008.02.013.">https://doi.org/10.1016/j.ajhg.2008.02.013.</a></p>
</div>
<div id="ref-Li2010">
<p>Li, Yongjin, and Jagdish C. Patra. 2010. â€œGenome-wide inferring gene-phenotype relationship by walking on the heterogeneous network.â€ <em>Bioinformatics</em> 26 (9): 1219â€“24. <a href="https://doi.org/10.1093/bioinformatics/btq108">https://doi.org/10.1093/bioinformatics/btq108</a>.</p>
</div>
<div id="ref-Pan2004">
<p>Pan, Jia-yu, Hyung Yang, Pinar Duygulu, and Christos Faloutsos. 2004. â€œAutomatic Multimedia Cross-modal Correlation Discovery,â€ 653â€“58.</p>
</div>
<div id="ref-Valdeolivas2018">
<p>Valdeolivas, Alberto, Laurent Tichit, Claire Navarro, Sophie Perrin, GaÃ«lle Odelin, Nicolas Levy, Pierre Cau, Elisabeth Remy, and AnaÃ¯s Baudot. 2018. â€œRandom walk with restart on multiplex and heterogeneous biological networks.â€ <em>Bioinformatics</em> 35 (3): 497â€“505. <a href="https://doi.org/10.1093/bioinformatics/bty637">https://doi.org/10.1093/bioinformatics/bty637</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Alberto Valdeolivas.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
